src/envoy/http/authn/http_filter_integration_test.cc

```diff
@@ -133,6 +133,48 @@ TEST_P(AuthenticationFilterIntegrationTest, CheckValidJwtPassAuthentication) {
   EXPECT_STREQ("200", response_->headers().Status()->value().c_str());
 }
 
+TEST_P(AuthenticationFilterIntegrationTest, CheckConsumedJwtHeadersAreRemoved) {
```

Comment:
It's a default behavior, so it's not necessary to crate separate test for this. You should add the check in all other tests to verify it is removed, regardless of request is accepted or rejected

Reply:
If the http request is rejected, the processing of the http request stops and the request will not be propagated to the downstream filters/backends. Thus, the removal is unnecessary as the entire http request is dropped.